clients/vault/src/system.rs

```diff
@@ -328,26 +332,29 @@ impl VaultService {
 		config: VaultServiceConfig,
 		shutdown: ShutdownSender,
 	) -> Result<Self, Error> {
-		let is_public_network = spacewalk_parachain.is_public_network().await;
-		let is_public_network = match is_public_network {
-			Ok(is_public_network) => is_public_network,
-			Err(error) => {
+		let is_public_network =
+			spacewalk_parachain.is_public_network().await.unwrap_or_else(|error| {
 				// Sometimes the fetch fails with 'StorageItemNotFound' error.
 				// We assume public network by default
 				tracing::warn!(
-					"Failed to fetch public network status from parachain: {}. Assuming public network.",
-					error
+					"Failed to fetch public network status from parachain: {error}. Assuming public network."
 				);
 				true
-			},
-		};
-
-		let stellar_vault_secret_key =
-			fs::read_to_string(&config.stellar_vault_secret_key_filepath)?;
-		let stellar_wallet = StellarWallet::from_secret_encoded(
-			&stellar_vault_secret_key.trim().to_string(),
-			is_public_network,
-		)?;
+			});
+
+		// check if both the config file and the network in the chain are the same.
```

Comment:
I think we did not check anywhere now that parachain and overlay_config refer to the same(both public or test) network?
Do we need this check to continue?  @ebma